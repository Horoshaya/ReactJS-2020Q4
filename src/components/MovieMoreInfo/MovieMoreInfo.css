.list {
  position: absolute;
  right: 0;
  top: 0;
  margin: 10px;
  padding: 17px 0;
  padding-top: 30px;
  width: 200px;
  min-height: 50px;
  background: #232323;
  color: #ffffff;
  list-style: none;
}

.closeIcon {
  position: absolute;
  right: 0;
  top: 0;
  padding: 7px;
}

.closeIcon:before {
  content: '\2715';
}

.moreButton {
  position: absolute;
  right: 0;
  display: block;
  margin: 10px;
  padding: 10px;
  padding-left: 15px;
  padding-right: 12px;
  border-radius: 50%;
  transition-duration: 0.3s;
  background: #232323;
  cursor: pointer;
}

.moreButton::after {
  content: '\2807';
}

.moreButton:hover {
  transform: scale(1.1);
  box-shadow: 0px 0px 6px 1px #0a0a0a;
}

.itemButton {
  padding: 10px;
  transition-duration: 0.3s;
  cursor: pointer;
}

.itemButton:hover {
  background: #f65261;
}
