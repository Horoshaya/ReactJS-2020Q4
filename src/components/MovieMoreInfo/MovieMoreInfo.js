import React, { Component } from 'react';
import styles from './MovieMoreInfo.css';

export default class MovieMoreInfo extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isDropdownOpen: false,
    };

    this.triggerModal = this.triggerModal.bind(this);
  }

  triggerModal() {
    this.setState(({ isDropdownOpen }) => ({
      isDropdownOpen: !isDropdownOpen,
    }));
  }

  render() {
    return (
      <>
        <div className={styles.moreButton} onClick={this.triggerModal}></div>

        {this.state.isDropdownOpen ? (
          <ul className={styles.list}>
            <span
              className={styles.closeIcon}
              onClick={this.triggerModal}
            ></span>
            <li
              className={styles.itemButton}
              onClick={() => this.props.triggerModal('modify')}
            >
              Edit
            </li>
            <li
              className={styles.itemButton}
              onClick={() => this.props.triggerModal('delete')}
            >
              Delete
            </li>
          </ul>
        ) : null}
      </>
    );
  }
}
// const MovieMoreInfoButton = ({ openDropdown, isDropdownOpen, openModal }) => {
//   return (
//     <>
//       <div className={styles.moreButton} onClick={openDropdown}></div>

//       {isDropdownOpen ? (
//         <ul className={styles.list}>
//           <span
//             className={styles.closeIcon}
//             onClick={showMoreInfoDropdown}
//           ></span>
//           <li className={styles.itemButton} onClick={() => openModal('edit')}>
//             Edit
//           </li>
//           <li className={styles.itemButton} onClick={() => openModal('delete')}>
//             Delete
//           </li>
//         </ul>
//       ) : null}
//     </>
//   );
// };

// export default MovieMoreInfoButton;
