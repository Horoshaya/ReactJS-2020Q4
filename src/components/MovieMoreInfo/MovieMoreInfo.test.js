import React from 'react';
import rerender from 'react-test-renderer';
import MovieMoreInfo from './MovieMoreInfo';
import { shallow } from 'enzyme';

describe('when MovieMoreInfo', () => {
  test('then snapshot created', () => {
    const component = rerender.create(<MovieMoreInfo />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('openModal', () => {
    const component = shallow(<MovieMoreInfo />);
    component.triggerModal;
    const closeIcon = component.find('div');
    closeIcon.simulate('click');
    expect(component.find('ul')).toBeTruthy();
  });
});
