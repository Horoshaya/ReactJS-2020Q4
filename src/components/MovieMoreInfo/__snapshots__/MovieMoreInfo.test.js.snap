// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MovieMoreInfo then snapshot created 1`] = `
<div
  onClick={[Function]}
/>
`;
